ui/slider.js

```diff
@@ -552,8 +552,27 @@ return $.widget( "ui.slider", $.ui.mouse, {
 	},
 
 	_calculateNewMax: function() {
-		var remainder = ( this.options.max - this._valueMin() ) % this.options.step;
-		this.max = this.options.max - remainder;
+		var precision = this._precision(),
+			multiplier = ( Math.pow( 10, precision ) ),
+			max = ( this.options.max * multiplier ),
+			min = ( this._valueMin() * multiplier ),
+			step =( this.options.step * multiplier ),
+			remainder = ( max - min ) % step;
+		this.max = parseFloat( ( ( max - remainder ) / multiplier ).toFixed( precision ) );
```

Comment:
Couldn't this follow the same logic as spinner and avoid the modulo operation? Something along the lines of:

```js
aboveMin = Math.floor( ( max - min ) / step ) * step;
max = aboveMin + min;
this.max = parseFloat( max.toFixed( this._precision() ) );
```